extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -411,6 +438,19 @@ export class AmpStory360 extends AMP.BaseElement {
     }
   }
 
+  /** @private */
+  restoreOrLoseGlContext_() {
+    if (this.renderer_) {
+      if (this.distance_ < MIN_WEBGL_DISTANCE) {
+        if (this.renderer_.gl.isContextLost()) {
+          this.lostGlContext_.restoreContext();
+        }
+      } else if (!this.renderer_.gl.isContextLost()) {
+        this.lostGlContext_.loseContext();
```

Comment:
Just curious: do you have some documentation on how this works? You get a "context lost" event and you have to forward it manually to the "WEBGL_lose_context" extension?

Reply:
[loseContext docs](https://developer.mozilla.org/en-US/docs/Web/API/WEBGL_lose_context/loseContext)
`loseContext` is a method on the WEBGL_lose_context extension.

`this.renderer_.gl.getExtension('WEBGL_lose_context')` is null after we call `loseContext` on it so we save a reference to it so we can call `restoreContext` on it. 

[Handling Lost Context in WebGL](https://www.khronos.org/webgl/wiki/HandlingContextLost)
We need to handle the `webglcontextlost` event so we can call `preventDefault`. Otherwise we can't recover the context.

Reply:
Super interesting thank you. It's surprisingly complicated